library/core/src/slice/mod.rs

```diff
@@ -3479,6 +3479,155 @@ impl<T> [T] {
+    ///
+    /// For a safe alternative see [`get_many`].
+    ///
+    /// # Safety
+    ///
+    /// Calling this method with overlapping or out-of-bounds indices is *[undefined behavior]*
+    /// even if the resulting references are not used.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(get_many)]
+    ///
+    /// let x = &[1, 2, 4];
+    ///
+    /// unsafe {
+    ///     assert_eq!(x.get_many_unchecked([0, 2]), [&1, &4]);
+    /// }
+    /// ```
+    ///
+    /// [`get_many`]: slice::get_many
+    /// [undefined behavior]: https://doc.rust-lang.org/reference/behavior-considered-undefined.html
+    #[unstable(feature = "get_many", issue = "none")]
+    #[inline]
+    pub unsafe fn get_many_unchecked<const N: usize>(&self, indices: [usize; N]) -> [&T; N] {
+        // NB: We need to go through a raw pointer to make miri happy
+        let slice: *const [T] = self;
+        let mut arr: mem::MaybeUninit<[&T; N]> = mem::MaybeUninit::uninit();
+        let arr_ptr = arr.as_mut_ptr();
+
+        // SAFETY: We expect `indices` to contain disjunct values that are
+        // in bounds of `self`.
+        unsafe {
+            for i in 0..N {
+                let idx = *indices.get_unchecked(i);
+                *(*arr_ptr).get_unchecked_mut(i) = &*slice.get_unchecked(idx);
+            }
+            arr.assume_init()
+        }
+    }
+
+    /// Returns multiple mutable references at once, without doing bounds checking.
+    ///
+    /// For a safe alternative see [`get_many_mut`].
+    ///
+    /// # Safety
+    ///
+    /// Calling this method with overlapping or out-of-bounds indices is *[undefined behavior]*
+    /// even if the resulting references are not used.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(get_many)]
+    ///
+    /// let x = &mut [1, 2, 4];
+    ///
+    /// unsafe {
+    ///     let [a, b] = x.get_many_unchecked_mut([0, 2]);
+    ///     *a *= 10;
+    ///     *b *= 100;
+    /// }
+    /// assert_eq!(x, &[10, 2, 400]);
+    /// ```
+    ///
+    /// [`get_many_mut`]: slice::get_many_mut
+    /// [undefined behavior]: https://doc.rust-lang.org/reference/behavior-considered-undefined.html
+    #[unstable(feature = "get_many", issue = "none")]
+    #[inline]
+    pub unsafe fn get_many_unchecked_mut<const N: usize>(
+        &mut self,
+        indices: [usize; N],
+    ) -> [&mut T; N] {
+        // NB: We need to go through a raw pointer to make miri happy
+        let slice: *mut [T] = self;
+        let mut arr: mem::MaybeUninit<[&mut T; N]> = mem::MaybeUninit::uninit();
+        let arr_ptr = arr.as_mut_ptr();
+
+        // SAFETY: We expect `indices` to contain disjunct values that are
+        // in bounds of `self`.
+        unsafe {
+            for i in 0..N {
+                let idx = *indices.get_unchecked(i);
+                *(*arr_ptr).get_unchecked_mut(i) = &mut *slice.get_unchecked_mut(idx);
+            }
+            arr.assume_init()
+        }
```

Comment:
```suggestion
        indices.map(|i| unsafe { &mut *slice.get_unchecked_mut(i) })
```

Reply:
- [`get_unchecked_mut()`](https://doc.rust-lang.org/std/primitive.slice.html#method.get_unchecked_mut) returns `&mut T`, so the `&mut *` would not be needed.
- This would not compile due to the static borrow checker for mutable borrows
- `i.map(|i| unsafe { &mut *(s.get_unchecked_mut(i) as *mut _) })` to make it raw pointers does not pass `miri` checks.